openlibrary/plugins/upstream/borrow.py

```diff
@@ -129,6 +129,18 @@ def POST(self, key):
         if not edition:
             raise web.notfound()
 
+        from openlibrary.book_providers import get_book_provider
+
+        # Direct to the first web book if at least one is available.
+        if (
+            action in ["borrow", "read"]
+            and (provider := get_book_provider(edition))
+            and (providers := provider.get_ebook_providers(edition))
+            and providers[0].access == "open-access"
+        ):
```

Comment:
This will also catch IA books! Will likely need a `if (provider := ... and provider.name != 'ia')`